src/app/child-dev-project/notes/model/note.ts

```diff
@@ -103,6 +104,7 @@ export class Note extends Entity {
    */
   removeChild(childId: string) {
     this.children.splice(this.children.indexOf(childId), 1);
+    this.children = [].concat(this.children); // TODO: hacky...
```

Comment:
```suggestion
    this.children = this.children.filter(c => c !== childId); 
```
this might make the purpose easier to understand